code/modules/clothing/glasses/glasses.dm

```diff
@@ -25,6 +28,36 @@ SEE_PIXELS// if an object is located on an unlit area, but some of its pixels ar
           // in a lit area (via pixel_x,y or smooth movement), can see those pixels
 BLIND     // can't see anything
 */
+
+/obj/item/clothing/glasses/mob_can_equip(mob/living/carbon/human/user, slot, disable_warning = 0)
+	var/mob/living/carbon/human/H = user
+	if(!istype(H) || stored_glasses || !glasses_fit)
+		return ..()
+	if(slot != slot_glasses)
+		return CANNOT_EQUIP
+	if(H.glasses)
+		stored_glasses = H.glasses
+		if(stored_glasses.w_class >= w_class)
+			if(!disable_warning)
+				to_chat(H, "<span class='danger'>You are unable to wear \the [src] as \the [H.glasses] are in the way.</span>")
+			stored_glasses = null
+			return CANNOT_EQUIP
```

Comment:
isn't this the exact situation we use CAN_EQUIP_BUT_SLOT_TAKEN for?

Reply:
We probably don't want the player to swap-out the goggles/magboots while there's still another pair of glasses/shoes in them.